app/models/host/base.rb

```diff
@@ -45,12 +45,18 @@ class Base < ApplicationRecord
     default_scope -> { where(taxonomy_conditions) }
 
     def self.taxonomy_conditions
-      org = Organization.expand(Organization.current)
-      loc = Location.expand(Location.current)
       conditions = {}
-      conditions[:organization_id] = Array(org).map { |o| o.subtree_ids }.flatten.uniq unless org.nil?
-      conditions[:location_id] = Array(loc).map { |l| l.subtree_ids }.flatten.uniq unless loc.nil?
-      conditions
+      if Organization.current.nil? && User.current.present? && !User.current.admin?
+        conditions[:organization_id] = User.current.organization_and_child_ids
```

Comment:
Method `organization_and_child_ids` uses Rails cache to cache the contents of results, that's probably why this is much faster.

Reply:
That is part of the reason, the other being that we don't call `subtree_ids` again for all the taxonomies when we already have all the subtrees.